growthstreet_zoo/zoo.py

```diff
@@ -117,17 +60,31 @@ def __str__(self):
 
     def __repr__(self):
         """Class representation, returned when class object is called."""
-        return '<%s: %s (%s)>' % (self.__class__.__name__, self.name, self.cage_contents)
+        return '<%s: %s (%s)>' % (self.__class__.__name__, self.name, self.contents)
 
     def add_animal(self, animal):
-        animal.get_competition_score()
-        animal.cage = self
-        self.cage_contents.append(animal)
+        """Add an instance of an animal to a Cage."""
+        self.contents.append(animal)
         self.predators_eat_prey()
 
     def predators_eat_prey(self):
-        response = '{} ate {}'.format(one_animal, another_animal)
-        return(response)
+        """Given animals in a cage. Check to see if any of them should be eaten."""
+        alive = [animal for animal in self.contents if animal.status == animal.STATUS_ALIVE]
+        permutations = itertools.permutations(alive, 2)
```

Comment:
This code isn't totally self explanatory, this would be a good place for a comment to explain what's going on.
Having said that, I think it might be that we could explore a simpler algorithm.  Let's discuss.